### PR TITLE
Add SlowCapturingScanner

### PR DESCRIPTION
**Pull request by Agryosha**
Add the new version of CapturingScanner based on NFA, which is more correct
than the previous one. Add new operators of 'non-greedy' repetitions, for example,
'+?', '??', '*?', which give a chance for users to tell, that the text, captured by
those repetitions, should be the shortest one
Added c++11 flag to configure.ac

### CODE REVIEW

File: pire/scanner_io.cpp [hunk 1]
```diff
@@ -96,6 +96,16 @@ void SlowScanner::Save(yostream* s) const
 				size += sizeof(unsigned) * i->size();
 			}
 		Impl::AlignSave(s, size);
+		if (need_actions)
+		{
+			size_t pos = 0;
+			for (yvector< yvector< Action > >::const_iterator i = m_actionsvec.begin(), ie = m_actionsvec.end(); i != ie; ++i)
+			if (!i->empty()) {
```
Comment (kv75) [inline pire/scanner_io.cpp]: Wrong formatting

**Comment by kv75** [comment]
Looks good to me.

File: tests/capture_ut.cpp [hunk 2]
```diff
@@ -50,6 +51,19 @@ SIMPLE_UNIT_TEST_SUITE(TestPireCapture) {
 		return fsm.Compile<Pire::CapturingScanner>();
 	}
 
+	SlowCapturingScanner SlowCompile(const char* regexp, int index, const Pire::Encoding& encoding = Pire::Encodings::Utf8())
+	{
+		Pire::Lexer lexer;
+		lexer.AddFeature(Pire::Features::Capture((size_t) index));
```
Comment (sergey-v-galtsev) [inline tests/capture_ut.cpp]: IMHO
`static_cast<size_t>(index)`

File: pire/fsm.cpp [hunk 3]
```diff
@@ -836,11 +836,11 @@ bool Fsm::LettersEquality::operator()(Char a, Char b) const
 	return true;
 }
 
-void Fsm::Sparse()
+void Fsm::Sparse(bool needEps /* = false */)
```
Comment (sergey-v-galtsev) [inline pire/fsm.cpp]: IMHO
`needEpsilons`

File: pire/fsm.h [hunk 4]
```diff
@@ -199,7 +199,7 @@ namespace Pire {
 
 
 		/// Builds letters equivalence classes
-		void Sparse();
+		void Sparse(bool needEps = false);
```
Comment (sergey-v-galtsev) [inline pire/fsm.h]: IMHO
`needEpsilons`

**Pull request merged**